plugin.video.cbc-sports/addon.py

```diff
@@ -0,0 +1,417 @@
+
+
+#99
+def play(url):
+	item = xbmcgui.ListItem(path=url)
+	return xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, item)
+
+
+def striphtml(data):
+	p = re.compile(r'<.*?>')
+	return p.sub('', data)
+
+
+def sanitize(data):
+	output = ''
+	for i in data:
+		for current in i:
+			if ((current >= '\x20') and (current <= '\xD7FF')) or ((current >= '\xE000') and (current <= '\xFFFD')) or ((current >= '\x10000') and (current <= '\x10FFFF')):
+			   output = output + current
+	return output
+
+
+
+def get_html(url):
+	req = urllib.request.Request(url)
+	req.add_header('User-Agent','User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux i686; rv:44.0) Gecko/20100101 Firefox/44.0')
+
+	try:
+		response = urllib.request.urlopen(req)
+		code = response.getcode()
+		cbclog = int(xbmcaddon.Addon().getSetting('cbclog'))
+		if cbclog >= 1:
+			xbmc.log('CBC Sports get_html code: ' + str(code), xbmc.LOGINFO)
+		if code == 403:              
+			xbmcgui.Dialog().notification(name, translation(30001), defaultimage, notetime, False)
+			sys.exit()
+		elif code == 22:              
+			xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+			sys.exit()	    
+		html = response.read()
+		response.close()
+	except urllib.error.URLError:              
+		xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+		sys.exit()	 
+	return html
+
+
+def get_params():
+	param = []
+	paramstring = sys.argv[2]
+	if len(paramstring) >= 2:
+		params = sys.argv[2]
+		cleanedparams = params.replace('?', '')
+		if (params[len(params) - 1] == '/'):
+			params = params[0:len(params) - 2]
+		pairsofparams = cleanedparams.split('&')
+		param = {}
+		for i in range(len(pairsofparams)):
+			splitparams = {}
+			splitparams = pairsofparams[i].split('=')
+			if (len(splitparams)) == 2:
+				param[splitparams[0]] = splitparams[1]
+
+	return param
+
+
+def addDir(name, url, mode, iconimage, fanart=False, infoLabels=True):
+	u = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + str(mode) + "&name=" + urllib.parse.quote_plus(name) + "&iconimage=" + urllib.parse.quote_plus(iconimage)
+	ok = True
+	liz = xbmcgui.ListItem(name)
+	liz.setInfo(type="Video", infoLabels={"Title": name})
+	liz.setProperty('IsPlayable', 'true')
+	if not fanart:
+		fanart=defaultfanart
+	liz.setArt({'thumb': iconimage, 'icon': iconimage, 'fanart': fanart})	
+	ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+	return ok
+
+
+def addDir2(name,url,duration,mode,iconimage, aired=False, plot=False, vheight=False, vwidth=False, fanart=False, infoLabels=False):
+	u=sys.argv[0]+"?url="+urllib.parse.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.parse.quote_plus(name)
+	ok=True
+	liz = xbmcgui.ListItem(name)
+	plot_text = ''
+	if plot:
+	    plot_text = plot
+	else:
+	    plot_text = name
+	aired_text = ''
+	if aired:
+            aired_text = aired
+	infoLabels={ "Title": name,
```

Comment:
This method of setting infolabels is deprecated in v20

Reply:
Noted.  Will modify once v20 is published.